src/Core/Commit.js

```diff
@@ -0,0 +1,314 @@
+/**
+ * @param {*} element
+ * @return {Element}
+ */
+function commitElement (element) {
+	if (element != null)
+		switch (element.constructor) {
+			case Element:
+				return element
+			case Array:
+				return elementFragment(element)
+			case String:
+			case Number:
+				return elementText(element)
+			case Function:
+			case Promise:
+				return createElement(element)
+			case Boolean:
```

Comment:
not sure I like Boolean and Symbol being turned into empty strings and not `elementUnknown`
also consider `elementUnknown` throwing as other vdom implementations do (https://github.com/Matt-Esch/virtual-dom)

Reply:
Booleans like null represent empty nodes so that `return props.x && <h1><h1>` would render nothing when the condition is false and the `h1` when it is true. Symbols should probably be handled by elementUnknown.

The elementUnknown logic is a bit different, it tries to render different nodes based on type i.e a node that looks like an iterable would fall into unknown, and elementUnknown would figure this out and render the iterable, other objects would get a `<pre>JSON.stringify()<pre>`  treatment, if JSON.stringify errors DIO will catch the error and cascade it up the tree to the nearest componentDidCatch implementation. Error Objects are rendered as `<details>stack trace...<details>`.

Reply:
The reason to throw the error is because there is no use case for rendering it. The user should never see a stack trace, and an error belongs in the console not the DOM. Same goes for JSON, you (almost) never actually want to render a JSON object to the DOM as a string; in almost every case it's a developer mistake.

Reply:
Yes it might make sense to throw instead of stringify on unrecognised Objects.

My line of reason for rendering Error instances was as for DX, for example: [https://jsbin.com/suyibaq/edit?js,output](https://jsbin.com/suyibaq/edit?js,output), just returns the error that propagated from render to view the details of the error in development without opening the console.